### PR TITLE
chore(logging): formatting fixes and improved directory creation

### PR DESCRIPTION
**Pull request by nsakaimbo**
### Description

Minor tweaking and cleanup of logger implementation:

- Rename file to `wpdesktop-main.log` (in case we want to differentiate from renderer/ui logs in the future)
- Use the [`make-dir`](https://github.com/sindresorhus/make-dir) package for more robust directory tree/parent directory creation. (Winston implementation is a simple `makedirSync` which is susceptible to `ENOENT` errors when the parent directory does not exist.) Tested on Win, Mac and Linux.
- Minor formatting tweaks to a couple log lines

### CODE REVIEW

**Comment by belcherj** [comment]
Where did we fall on making all logging string interpolation? "log.info( `New update is available: ${info.version}` )"

**Comment by nsakaimbo** [comment]
> Where did we fall on making all logging string interpolation?

`log.info( `New update is available: ${info.version}` )`

Is the above in the code somewhere? (Can't find it.) 

TL;DR we refactored so tokenizers (`%s, %o`, etc) aren't necessary (and have been completely removed from the code). String interpolation is fine, if someone want to use it for any reason.

**Comment by belcherj** [comment]
In this PR you changed a couple of string concats to string interpolation:

https://github.com/Automattic/wp-desktop/pull/815/files#diff-1da2efbbb894f684c5f8fd05c6d805ceR25

**Comment by belcherj** [comment]
Wait, what is the difference between status and error in:

`log.info( 'Port check failed - ' + status, error );`

I think I misunderstood how what the comma is doing here.
